docs/changelog/2300.md

```diff
@@ -0,0 +1 @@
+- Removed support for configuring preCICE as static library.
```

Comment:
I think we should add a comment that static was anyway not working before, thus we bring this in a minor release.

Reply:
Static libs are broken since we use the object library for preciceCore, so 3.0.0.

```suggestion
- Removed support for configuring preCICE as static library, which has been broken since version 3.0.0.
```